### PR TITLE
[CPDEV-101744] Update helm, python dependencies

### PR DESCRIPTION
**Pull request by ilia1243**
### Description
* Drop support python 3.7
    * Code changes to respect minimal version of python 3.8
* Update helm v3.14.4
* Added dependency on useful-types.
* Updated python dependencies
    * Updated ansible to 7.7.0
    * Fixes in schema.py to support new jsonschema version.
* Updated go version for ipip_check.

### Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] Integration CI passed
- [x] Unit tests. If Yes list of new/changed tests with brief description
- [x] There is no merge conflicts

#### Unit tests
Changed tests to respect changed JSON schema validation errors.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -39,29 +38,31 @@ dependencies = [
...
+    "referencing==0.35.*",
+    "typing_extensions==4.11.*",
+    "pathvalidate==3.2.*",
+    "useful_types==0.2.*",
 ]
-requires-python = ">=3.7"
+requires-python = ">=3.8"
 
 [project.optional-dependencies]
 ansible = ["ansible==7.0.*"]
```
Comment (ilia1243) [inline pyproject.toml:56]: Ansible is outdated for 1.5 years, and not updated without external request due to potentially high impact.

**Review by alexarefev — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (koryaga) [reply pyproject.toml:56]: what about 7.X updated version ? https://pypi.org/project/ansible/7.7.0/

Comment (ilia1243) [reply pyproject.toml]: https://github.com/Netcracker/KubeMarine/pull/661/commits/76c7c04b4847ca92d0dfd20fdf00877aee7bb1b2

**Review by koryaga — APPROVED** [review]

**Comment by koryaga** [comment]
@OleksandrIaresko

**Pull request merged**